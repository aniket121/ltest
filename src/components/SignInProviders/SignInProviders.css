.paper {
  width: 320px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.providers {
  display: flex;
  justify-content: space-between;
  padding: 0 44px;
  width: 100%;
  box-sizing: border-box;
}

.message {
  text-transform: uppercase;
}

.GoogleLogin {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  font-family: 'F37 Ginger', sans-serif;
  width: 56px;
  height: 56px;
  background-image: linear-gradient(
    -208deg,
    rgba(223, 73, 48, 1) 0%,
    rgba(170, 42, 21, 1) 100%
  );
  color: #fff;
  border-radius: 50%;
  border: none;
  cursor: pointer;
}

.GoogleLogin:focus {
  outline: none;
}

.FacebookLogin {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'F37 Ginger', sans-serif;
  width: 56px;
  height: 56px;
  background-image: linear-gradient(
    -212deg,
    rgba(80, 124, 192, 1) 0%,
    rgba(2, 68, 171, 1) 100%
  );
  color: #fff;
  border-radius: 50%;
  border: none;
  cursor: pointer;
}

.FacebookLogin:focus {
  outline: none;
}

.EmailLogin {
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
    -34deg,
    rgba(0, 39, 163, 1) 0%,
    rgba(132, 0, 160, 1) 35%,
    rgba(225, 69, 59, 1) 76%,
    rgba(255, 205, 22, 1) 100%
  );
  width: 56px;
  height: 56px;
  border-radius: 50%;
  color: #fff;
  font-size: 28px;
  font-style: normal;
  cursor: pointer;
}

.Email::selection {
  background-color: transparent;
}

.providerWrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.providerName {
  text-transform: uppercase;
  color: rgba(128, 128, 128, 1);
  font-size: 12px;
}
.Email {
  font-style: normal;
  margin-bottom: 8px;
}

.providerIcon {
  font-style: normal;
}

.providerIcon::before {
  font-size: 24px;
}
