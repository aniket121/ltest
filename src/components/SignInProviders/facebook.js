import React from 'react';
import styles from './SignInProviders.css';

const FacebookLogin = props => (
  <div
    className={styles.FacebookLogin}
    onClick={props.onClick}
  >
    <i className={"fa fa-facebook " + styles.providerIcon}></i>
  </div>
);

export default FacebookLogin;
