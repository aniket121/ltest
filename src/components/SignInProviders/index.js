import React from 'react';
import FacebookLogin from 'react-facebook-login';
import MyFacebookLogin from './facebook';
import GoogleLogin from 'react-google-login';
import config from '../../config';
import styles from './SignInProviders.css';

const SIGN_IN_PROVIDERS = ['facebook', 'google', 'email'];

const EmailLogin = ({ handleEmailLogin }) =>
  <div className={styles.EmailLogin} onClick={handleEmailLogin}>
    <i className={styles.providerIcon + ' ' + styles.Email}>@</i>
  </div>;

export const Provider = ({
  provider,
  handleGoogleLogin,
  handleFacebookLogin,
  handleEmailLogin
}) => {
  let childProvider, providerName;
  switch (provider) {
    case 'google':
      childProvider = (
        <GoogleLogin
          className={styles.GoogleLogin}
          clientId={config.youtube.CLIENT_ID}
          onSuccess={handleGoogleLogin}
        >
          <i className={styles.providerIcon + ' fa fa-google-plus'} />
        </GoogleLogin>
      );
      providerName = 'Google+';
      break;
    case 'facebook':
      childProvider = (
        <FacebookLogin
          appId={config.facebook.APP_ID}
          textButton=""
          fields="name,email,picture"
          cssClass={styles.FacebookLogin}
          callback={handleFacebookLogin}
          icon={'fa-facebook ' + styles.providerIcon}
        />
      );
      providerName = 'Facebook';
      break;
    case 'email':
      childProvider = <EmailLogin handleEmailLogin={handleEmailLogin} />;
      providerName = 'Email';
      break;
    default:
      childProvider = null;
  }

  return (
    <div className={styles.providerWrapper}>
      {childProvider}
      <p className={styles.providerName}>
        {providerName}
      </p>
    </div>
  );
};

const SignInProviders = props =>
  <div className={styles.paper}>
    <p className={styles.message}>
      {props.message}
    </p>
    <div className={styles.providers}>
      {SIGN_IN_PROVIDERS.map(provider =>
        <Provider key={provider} provider={provider} {...props} />
      )}
    </div>
  </div>;

export default SignInProviders;
