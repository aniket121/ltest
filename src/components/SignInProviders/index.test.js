import React from 'react';
import { shallow, mount } from 'enzyme';
import SignInProviders, { Provider } from 'components/SignInProviders';

const createProps = settings => ({
  message: 'Log in with',
  ...settings
});

describe('<SignInProviders />', () => {
  test('renders with default props', () => {
    const props = createProps();
    const wrapper = shallow(<SignInProviders {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with all three providers', () => {
    const props = createProps();
    const wrapper = shallow(<SignInProviders {...props} />);
    expect(wrapper.find(Provider)).toHaveLength(3);
  });
})
