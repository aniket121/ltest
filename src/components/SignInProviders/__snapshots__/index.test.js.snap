exports[`<SignInProviders /> renders with default props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <p
        className={undefined}>
        Log in with
    </p>
    <div
        className={undefined}>
        <Unknown
            message="Log in with"
            provider="facebook" />
        <Unknown
            message="Log in with"
            provider="google" />
        <Unknown
            message="Log in with"
            provider="email" />
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <p
            className={undefined}>
            Log in with
      </p>
      <div
            className={undefined}>
            <Unknown
                  message="Log in with"
                  provider="facebook" />
            <Unknown
                  message="Log in with"
                  provider="google" />
            <Unknown
                  message="Log in with"
                  provider="email" />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SignInProviders
        message="Log in with" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "message": "Log in with",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <p
                    className={undefined}>
                    Log in with
          </p>
          <div
                    className={undefined}>
                    <Unknown
                              message="Log in with"
                              provider="facebook" />
                    <Unknown
                              message="Log in with"
                              provider="google" />
                    <Unknown
                              message="Log in with"
                              provider="email" />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <p
                    className={undefined}>
                    Log in with
          </p>
          <div
                    className={undefined}>
                    <Unknown
                              message="Log in with"
                              provider="facebook" />
                    <Unknown
                              message="Log in with"
                              provider="google" />
                    <Unknown
                              message="Log in with"
                              provider="email" />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignInProviders
    message="Log in with" />,
}
`;
